ziti-tunnel/docker/docker-entrypoint.sh

```diff
@@ -32,20 +32,29 @@ fi
 
 json="${persisted_dir}/${NF_REG_NAME}.json"
 if [ ! -f "${json}" ]; then
-    echo "identity configuration ${json} does not exist"
-    for dir in "/var/run/secrets/netfoundry.io/enrollment-token" "${persisted_dir}"; do
-        _jwt="${dir}/${NF_REG_NAME}.jwt"
-        echo "looking for ${_jwt}"
-        if [ -f "${_jwt}" ]; then
-            jwt="${_jwt}"
-            break
-        fi
+    echo "WARN: identity configuration ${json} does not exist" >&2
+    for dir in  "/var/run/secrets/kubernetes.io/enrollment-token" \
+                "/enrollment-token" \
+                "${persisted_dir}";
+        do
+            [[ -d ${dir} ]] || {
+                echo "WARN: ${dir} is not a directory" >&2
+                continue
+            }
+            _jwt="${dir}/${NF_REG_NAME}.jwt"
+            echo "INFO: looking for ${_jwt}"
+            if [ -f "${_jwt}" ]; then
+                jwt="${_jwt}"
+                break
+            else
+                echo "WARN: failed to find ${_jwt} in ${dir}" >&2
+            fi
     done
-    if [ -z "${jwt}" ]; then
-        echo "ERROR: ${NF_REG_NAME}.jwt was not found in the expected locations"
+    if [ -z "${jwt:-}" ]; then
```

Comment:
Is this a default parameter expansion with an empty default? I'm curious how is this different from expanding the variable without using expansion?

Reply:
Yes, with BASH option no unset in force it is an error to interpolate ${jwt} when unset, and so the script exits prematurely.

Reply:
That is, when both nounset and errexit are set, which they are, intentionally.

Reply:
I see. Good catch!